tests/unit/executors/indexers/test_binary_indexer.py

```diff
@@ -30,3 +30,105 @@ def validate(req):
         d.ClearField('embedding')
     with f:
         f.search(docs_no_embedding, output_fn=validate, override_doc_id=False)
+
+
+def test_binarypb_update1(test_metas):
+    with BinaryPbIndexer(metas=test_metas) as idxer:
+        idxer.add([1, 2, 3], [b'oldvalue', b'same', b'random'])
+        idxer.save()
+        assert idxer.size == 3
+        first_size = os.fstat(idxer.write_handler.body.fileno()).st_size
+        save_abspath = idxer.save_abspath
+
+    with BaseIndexer.load(save_abspath) as idxer:
+        assert idxer.query(1) == b'oldvalue'
+
+    with BaseIndexer.load(save_abspath) as idxer:
+        # no update triggered AT ALL when encountering missing key
+        # atomic op. at indexer level
+        with pytest.raises(KeyError):
+            idxer.update([1, 2, 99], [b'newvalue', b'same', b'decoy'])
+
+        idxer.save()
+
+    with BaseIndexer.load(save_abspath) as idxer:
+        assert idxer.query(1) == b'oldvalue'
+        second_size = os.fstat(idxer.query_handler._body.fileno()).st_size
+        assert second_size == first_size
+
+    with BaseIndexer.load(save_abspath) as idxer:
+        # some new value
+        idxer.update([1, 2], [b'newvalue', b'same'])
+        idxer.save()
+        third_size = os.fstat(idxer.write_handler.body.fileno()).st_size
+        assert third_size > first_size
+        assert idxer.size == 3
+
+    with BaseIndexer.load(save_abspath) as idxer:
+        assert idxer.query(1) == b'newvalue'
+        assert idxer.query(2) == b'same'
+        assert idxer.query(3) == b'random'
+        assert idxer.query(99) is None
+
+
+def test_binarypb_add_and_update_not_working(test_metas):
+    with BinaryPbIndexer(metas=test_metas) as idxer:
+        idxer.add([11, 12], [b'eleven', b'twelve'])
+        idxer.save()
+        # FIXME `add` and `update` won't work in the same context
```

Comment:
this may be fixed with the lazy flush?

Reply:
I think add and update should be supported; if we merge this please create a ticket to keep track of this

Reply:
Yes, I will

Reply:
https://github.com/jina-ai/jina/issues/1413